gdb/rust-lang.c

```diff
@@ -1183,7 +1415,139 @@ rust_evaluate_subexp (struct type *expect_type, struct expression *exp,
 	  }
       }
       break;
+    case STRUCTOP_ANONYMOUS:
+      {
+        /* Anonymous field access, i.e. foo.1 */
+        struct value *lhs;
+        int pc, field_number, nfields;
+        struct type *type, *variant_type;
+        struct disr_info disr;
+
+        pc = (*pos)++;
+        field_number = longest_to_int (exp->elts[pc + 1].longconst);
+        (*pos) += 2;
+        lhs = evaluate_subexp (NULL_TYPE, exp, pos, noside);
+
+        type = value_type (lhs);
+        if (TYPE_CODE (type) == TYPE_CODE_UNION) {
+          struct cleanup *cleanup;
+
+          disr = rust_get_disr_info (type, value_contents(lhs),
+                                     value_embedded_offset(lhs),
+                                     value_address(lhs), lhs);
+
+
+          if (disr.field_no < 0) {
+            error (_("Could not determine variant of enum value"));
+          }
+
+          cleanup = make_cleanup (xfree, disr.name);
+
+          variant_type = TYPE_FIELD_TYPE(type, disr.field_no);
+          nfields = TYPE_NFIELDS (variant_type);
+
+          // note that there is the extra discriminant field as well,
+          // so the actual field number is field_number + 1
+          if (field_number >= nfields - 1 || field_number < 0) {
+            // unsure how to call error with a cleanup
```

Comment:
`error` will throw and will invoke the cleanups that were installed between this spot on the stack and the spot where the catch is done.  So there's nothing you need to do here.